Update ProcessInstanceHelper.java to fix runtime bundle executing wrong start event when start process instance by message [issue #4382]

abhinay9955 opened the pull request:
Update ProcessInstanceHelper.java to fix runtime bundle in activiti cloud executing wrong start event when start process instance by message [issue #4382]

abhinay9955 committed:
removed_conflicting_condition

abhinay9955 committed:
Revert "removed_conflicting_condition"

This reverts commit 2c5eb4340b01d96dc366c4714a31ed3457f56ed0.

abhinay9955 committed:
fix start Process By Message

abhinay9955 committed:
Update ProcessInstanceHelper.java

Fix Start Process Instance by Message

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Activiti/Activiti?pullRequest=4383) <br/>All committers have signed the CLA.

abhinay9955 commented:
Hi @miguelruizdev , can you please review and approve workflow runs
Thank you !!

miguelruizdev commented:
Hi @abhinay9955, we would need test coverage in order to accept the Pull Request.
Also, it'd be great if you add @igdianov as a reviewer, since I see he's the author of the code you are deleting and probably he has a greater context on that.

abhinay9955 commented:
Sure @miguelruizdev , Thanks!! I could not find any test class for ProcessInstanceHelper.java , that's why didnt added tests.  Will add if @igdianov asks fo it. I hope this is fine  :)

igdianov reviewed (changes requested):
@abhinay9955 I am a concerned that this is going to be a breaking change, ie in APS, that changes the behavior of using a model message ref to find and set message ref to message name by matching it to message id in case the message name is not matching message ref. 

We need to consult with @VitoAlbano who added this code as part of this commit: https://github.com/Activiti/Activiti/commit/6ddaae94ece697dc45fdd0c08ee28d62dda56f96

abhinay9955 committed:
changes according to @igdianov's comments

abhinay9955 committed:
fix merge conflict

abhinay9955 committed:
changes according to @igdianov's comments

Added code for condition when there is no message reference

abhinay9955 commented:
Hi @igdianov , i have updated the code , i think this change addresses your concern , Thanks :)

igdianov commented:
> Hi @igdianov , i have updated the code , i think this change addresses your concern , Thanks :)

Can you add test coverage for it, please? Thanks!

abhinay9955 committed:
Added unit tests

abhinay9955 committed:
Update ProcessInstanceHelper.java

added getInitialFlowByMessage to seperate the logic of getting initial flow from start processInstance method

abhinay9955 committed:
Create ProcessInstanceHelperTest.java

Added unit test

abhinay9955 committed:
Create StartProcessInstanceByMessage.testInitialFlowElementByMessage.bpmn20.xml

Added resource for testing

abhinay9955 committed:
Create StartProcessInstanceByMessage.testNoMessagePresent.bpmn20.xml

abhinay9955 committed:
Create StartProcessInstanceByMessage.testOnlyOneMessageExist.bpmn20.xml

abhinay9955 commented:
Hi @igdianov , added unit tests please review . Thanks :)
Sorry for delay.

abhinay9955 committed:
Update ProcessInstanceHelperTest.java

abhinay9955 committed:
update tests

abhinay9955 committed:
Merge branch 'abhinay9955-4382-fixStartEventWithMessage' of https://www.github.com/abhinay9955/Activiti into abhinay9955-4382-fixStartEventWithMessage

abhinay9955 committed:
Refactored tests

abhinay9955 committed:
removed import warning

abhinay9955 commented:
@VitoAlbano please review !! Thanks !!

abhinay9955 commented:
Hi @VitoAlbano @igdianov @miguelruizdev , i have added test cases long ago , can you please review .
Thanks in Advance!!

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/Activiti/Activiti?pullRequest=4383) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/Activiti/Activiti?pullRequest=4383) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/Activiti/Activiti?pullRequest=4383) it.</sub>
